packages/valory/skills/register_termination_abci/models.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the shared state for the register-termination ABCI application."""
+
+from typing import Any
+
+from packages.valory.skills.abstract_round_abci.models import (
+    BenchmarkTool as BaseBenchmarkTool,
+)
+from packages.valory.skills.abstract_round_abci.models import Requests as BaseRequests
+from packages.valory.skills.abstract_round_abci.models import (
+    SharedState as BaseSharedState,
+)
+from packages.valory.skills.register_termination_abci.composition import (
+    RegisterTerminateAbciApp,
+)
+from packages.valory.skills.termination_abci.models import TerminationParams
+from packages.valory.skills.transaction_settlement_abci.models import (
+    RandomnessApi as TransactionSettlementRandomness,
+)
+from packages.valory.skills.transaction_settlement_abci.models import TransactionParams
+
+
+Requests = BaseRequests
+BenchmarkTool = BaseBenchmarkTool
+RandomnessApi = TransactionSettlementRandomness
+
+
+class SharedState(BaseSharedState):
+    """Keep the current shared state of the skill."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the state."""
+        super().__init__(*args, abci_app_cls=RegisterTerminateAbciApp, **kwargs)
+
+
+class Params(TerminationParams, TransactionParams):
```

Comment:
If you apply comments in `packages/valory/skills/termination_abci/models.py` then we don't have a direct dependency on `transaction_settlement_abci` anymore here. That's highly desirable as it makes it much easier to compose (need to know less about the internals). The whole skill then also no longer has a direct dependency on `transaction_settlement_abci`